core/process.py

```diff
@@ -1461,6 +1461,8 @@ def WaitForOne(self, eintr_retry):
       # Print newline after Ctrl-C.
       if term_sig == signal_.SIGINT:
         print('')
+        if pid == self.job_state.last_stopped_pid:
+            self.job_state.last_stopped_pid = -1
```

Comment:
Thanks for looking at this!  I think the general idea of resetting last_stopped_pid is right, but probably not on Ctrl-C in general?

Note that there is only one instance of `JobState` in the whole interpreter.   I should probably rename it because it is confusing with `Job`.  So it isn't a 1:1 relationship.

Let me try to think of a test case where this wouldn't work... I think if you do something besides Ctrl-C it won't work.

Like you could resume `cat`, and instead of `Ctrl-C`, you can do `Ctrl-\`, or kill it from outside the shell.  Does the bug still happen then?

Reply:
okay, I'll check... i am not quite sure how `SIGINT` propagates through the code, but this branch seems to be checking for `SIGINT` specfically, `KeybardInterrupt` is explicitly ignored above...

Reply:
Seems to work if I use kill from inside the shell...

```
osh-0.9.5$ cat                                                                                                                                                 
  C-c C-z                                                                                                                                                      
[PID 4969] Stopped
osh-0.9.5$ kill -INT 4969                                                                                                                                      
osh-0.9.5$ fg                                                                                                                                                  
Continue PID 4969                                                                                                                                                                                                                                                                                      
osh-0.9.5$      
```


```
osh-0.9.5$ cat                                                                                                                                                 
  C-c C-z                                                                                                                                                      
[PID 6032] Stopped                                                                                                                                             
osh-0.9.5$ fg                  # ran kill -INT 4969 from outside the shell                                                                                                                                
Continue PID 6032                                                                                                                                                                                                                                                                                          
osh-0.9.5$  
```

This case fails:
```
osh-0.9.5$ cat                                                                                                                                                 
  C-c C-z                                                                                                                                                      
[PID 6077] Stopped                                                                                                                                             
osh-0.9.5$ kill -9 6077                                                                                                                                        
osh-0.9.5$ fg                                                                                                                                                  
Continue PID 6077                                                                                                                                              
osh I/O error: No such process                                                                                                                                 
brandon@rizzo:~/src/oil$
```
So I guess it does need to be more general.